diff --git a/DAG/inter_profile.cpp b/DAG/inter_profile.cpp
index dd7e69a..6412d8c 100644
--- a/DAG/inter_profile.cpp
+++ b/DAG/inter_profile.cpp
@@ -149,7 +149,7 @@ static void create_controller(Function *F,
     // cmp = Builder.CreateAnd(cmp, calls[i]);
   }
 
-  auto *br = Builder.CreateCondBr(cmp, C->getLoopPreheader(), L->getLoopPreheader());
+  auto *br = Builder.CreateCondBr(cmp, L->getLoopPreheader(), L->getLoopPreheader());
   // add_dump_msg(C->getLoopPreheader(), "going to clone\n");
   // add_dump_msg(C->getLoopPreheader(), F->getName());
   // add_dump_msg(L->getLoopPreheader(), "going to original loop\n");
diff --git a/DAG/intra_profile.h b/DAG/intra_profile.h
index 065b0ea..9dcc142 100644
--- a/DAG/intra_profile.h
+++ b/DAG/intra_profile.h
@@ -217,7 +217,7 @@ void create_BBControl(Function *F,
   // if c2 > gap then we change switch_control to jump to BBOpt, otherwise,
   // jump to BB
   Value *gap_cmp = Builder.CreateICmpSGE(c2, gap, "gap.cmp");
-  Value *new_target = Builder.CreateSelect(gap_cmp, BBOpt_target_value, BB_target_value);
+  Value *new_target = Builder.CreateSelect(gap_cmp, BB_target_value, BB_target_value);
 
   // decide if it is time to change the switch jump
   Value *iter_cmp = Builder.CreateICmpEQ(c1, n_iter, "iter.cmp");
